src/lib/nvme_pt_ioctl.c

```diff
@@ -723,14 +723,15 @@ static int opal_start_anybody_session(int fd, struct opal_device *dev, int sp)
 		dev->session.tsn = dev->payload.tokens[5]->vals.uint;
 	} else {
 		SEDCLI_DEBUG_MSG("Response parsed incorrectly.\n");
-		return ret;
+		goto put_tokens;
 	}
 
 	if (dev->session.hsn == 0 && dev->session.tsn == 0) {
```

Comment:
how this condition can be true if we can only get to this if statement when opal_snd_rcv_cmd_parse_chk returns OPAL_SUCCESS?
I think this whole if may not be needed